realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -1056,6 +1057,52 @@ public void compactRealm_onExternalStorage() {
         Realm.deleteRealm(config);
     }
 
+    private Pair<Long, Long> populateTestRealmAndCompactOnLaunch(CompactOnLaunchCallback compactOnLaunch) {
+        final String REALM_NAME = "test.realm";
+        RealmConfiguration realmConfig = configFactory.createConfiguration(REALM_NAME);
+        Realm realm = Realm.getInstance(realmConfig);
+        populateTestRealm(realm, 100);
+        realm.beginTransaction();
+        realm.deleteAll();
+        realm.commitTransaction();
+        realm.close();
+        long before = new File(realmConfig.getPath()).length();
+        RealmConfiguration.Builder builder = configFactory.createConfigurationBuilder();
+        File folder = configFactory.getRoot();
```

Comment:
Please use the same pattern we do elsewhere. Also, this is already being set in the configFactory, so

```
RealmConfiguration realmConfig = configFactory.createConfigurationBuilder()
     .name(REALM_NAME)
     .compactOnLaunch(compactOnLaunch)
     .build();
```